kitsune/wiki/utils.py

```diff
@@ -182,26 +182,53 @@ def get_visible_document_or_404(
     except Document.DoesNotExist:
         pass
 
-    if (
-        not look_for_translation_via_parent
-        or not (locale := kwargs.get("locale"))
-        or (locale == settings.WIKI_DEFAULT_LANGUAGE)
-    ):
-        # We either don't want to try to find the translation via its parent, or it doesn't
-        # make sense, because we're not making a locale-specific request or the locale we're
-        # requesting is already the default locale.
+    if not (locale := kwargs.get("locale")):
         raise Http404
 
-    # We couldn't find a visible translation in the requested non-default locale, so let's
-    # see if we can find a visible translation via its parent.
+    slug = kwargs.get("slug")
+
+    if slug and look_for_translation_via_parent:
+        # Find documents with this slug in other locales
+        other_docs_qs = (
+            Document.objects.filter(slug=slug).exclude(locale=locale).select_related("parent")
```

Comment:
Shouldn't we use `visible` here too like above instead of `filter`?

Reply:
My thoughts were that since we are looking for a path to a document through other documents, the documents we look at to move to a valid document should all be collected. We check for visibility later before presenting a final document. What if a document is restricted in one language but not in another?

Reply:
That's probably problematic and it shouldn't happen. The restricted visibility should be the same across all locales

Reply:
I still believe this is the right approach - doing discovery of the redirect chain has nothing to do with visibility. If  there is a case where a document in the chain is not visible to the user, but the document the user should land on is visible - then they won't get there if we change this to use `.visible`. 
We could wind up with a case where locale leaders or content editors or others with unique permissions will get different redirects than other users. 
Unless you can think of a reason that this is harmful in some other way, I'd like to leave it.